src/lineup/internal/panel/PanelTabContainer.ts

```diff
@@ -1,64 +1,124 @@
 import {PanelTab} from './PanelTab';
+import {PanelNavButton} from './PanelButton';
 
-export default class PanelTabContainer {
-
-  readonly node: HTMLElement;
 
-  private tabs: PanelTab[] = [];
 
-  private currentTab: PanelTab;
+/**
+ * Acts as the navigation to the TabPanels
+ * Allows the addition in the correct order of PanelNavButtons
+ */
+class PanelTabHeader {
+  public node: HTMLElement;
+  public navButtons: PanelNavButton[] = [];
 
+  /**
+   * @param parent The parent HTML DOM element
+   */
   constructor(parent: HTMLElement) {
-    this.node = parent.ownerDocument.createElement('main');
-    this.node.classList.add('tab-content');
+    this.node = parent.ownerDocument.createElement('ul');
+    this.node.className = 'nav nav-tabs';
     parent.appendChild(this.node);
   }
 
-  private get defaultTab(): PanelTab {
-    return this.tabs[0];
+  /**
+   *
+   * @param navTabs NavTabs array sorted according to their order property
+   * @param nav New nav being added to the NavTabs
+   */
+  getNextSibling(sortedNavTabs: PanelNavButton[], nav: PanelNavButton): PanelNavButton | null {
+    const navTabsLength = sortedNavTabs.length;
+    const navIndex = sortedNavTabs.findIndex((n) => n.order === nav.order);
+    return navIndex === navTabsLength - 1 ? null : sortedNavTabs[navIndex + 1];
   }
 
-  addTab(tab: PanelTab) {
-    this.tabs = [...this.tabs, tab];
-    this.node.appendChild(tab.node);
+  /**
+   * Add nav-tab to the nav-tabs in the correct order.
+   * @param button PanelNavButton instance to add
+   */
+  addNav(nav: PanelNavButton) {
+    const sameOrderNav = this.navButtons.find((n) => n.order === nav.order); // find nav with same order
+    this.navButtons = [...this.navButtons, nav].sort((nav1, nav2) => nav1.order - nav2.order);
+    this.node.appendChild(nav.node);
+    const nextSibling = sameOrderNav || this.getNextSibling(this.navButtons, nav); // if same order property append new nav before it
+    if (nextSibling) {
+      this.node.insertBefore(nav.node, nextSibling.node);
+    } else {
+      this.node.appendChild(nav.node);
+    }
```

Comment:
We can shift the sorting to the extension point, as we only want to sort the panel tabs that are added by the extension point. That would simplify the code here.